tests/kubevirtresource.go

```diff
@@ -0,0 +1,188 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package tests
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+	"kubevirt.io/kubevirt/tests/flags"
+	"kubevirt.io/kubevirt/tests/framework/checks"
+	"kubevirt.io/kubevirt/tests/util"
+)
+
+var (
+	KubeVirtDefaultConfig v1.KubeVirtConfiguration
+	originalKV            *v1.KubeVirt
+)
+
+func AdjustKubeVirtResource() {
```

Comment:
IMO this is more internal/testsuite related.

Reply:
I think it needs to be referenced to something explicit. The name hints it is kubevirt CR related, but it may be indeed misleading.
I do not have anything better for now and "internal" sounds a too generic tells-me-nothing name, no different then `utils`, `common` and friends.

Reply:
By internal/testsuite I was trying to say it is not meant to be used outside of suite setup. Seeing this as it suggests to me I can use it in tests.

Reply:
Aha... that is a really good point.
To "hide" the helpers that should not be used in tests and make it clear.

Do you have a general suggestion that we can implement in a consistent manner?
Is it to create an `internal` folder to enforce not using there helpers outside? If this is what you mean, I think we may need to restructure the `tests` folder a bit differently to support something like that.

Reply:
I would need to have a look as we moved a lot right now. I think [this](https://github.com/kubevirt/kubevirt/pull/7188)  is something that can also help us in some of the areas. Would you want to have a look at the PR and provide feedback?

Reply:
I saw only the [first](https://github.com/kubevirt/kubevirt/pull/7141) attempt of adding the "framework", inspired from k8s.
At the time, I suggested to start a design discussion for brainstorming and see on which points we can reach agreement.

I will have a look at this latest suggestion, thank you for the ref.
I still think it is worth a design stage.